storage/innobase/handler/ha_innodb.cc

```diff
@@ -8718,7 +8718,7 @@ ha_innobase::update_row(
 	}
 
 #ifdef WITH_WSREP
-	if (error == DB_SUCCESS && trx->is_wsrep()
+	if (error == DB_SUCCESS && wsrep_on(m_user_thd)
 	    && wsrep_thd_is_local(m_user_thd)
 	    && !wsrep_thd_ignore_table(m_user_thd)) {
```

Comment:
This is replacing a check of `trx->wsrep` with function calls, which introduces a performance regression to all successful `UPDATE` operations. Can we fix the way how `trx->wsrep` is initialized?

Also, could the multiple `wsrep_thd_` function calls be consolidated into one?

The same anti-pattern occurs also in `ha_innobase::delete_row()`, affecting `DELETE` operations.